ansible_rulebook/engine.py

```diff
@@ -217,14 +220,45 @@ async def start_source(
         )
 
 
+class RulebookFileChangeHandler(FileSystemEventHandler):
+    modified = False
+
+    def on_modified(self, event):
+        logger.debug(f"Rulebook file {event.src_path} has been modified")
+        self.modified = True
+
+    def is_modified(self):
+        return self.modified
+
+
+async def monitor_rulebook(rulebook_file):
+    event_handler = RulebookFileChangeHandler()
+    to_observe = os.path.abspath(rulebook_file)
+    observer = Observer()
+    observer.schedule(event_handler, to_observe, recursive=True)
+    observer.start()
+    try:
+        while not event_handler.is_modified():
+            await asyncio.sleep(1)
+    finally:
```

Comment:
@tarilabs when we cancel tasks because the rulebook is ending normally the monitor task is going to get cancelled,  in that case should we be executing this finally clause which will raise the execption HotReloadException, should the HotReloadException have a check to ensure that the event_handler.is_modified is True before raising the exception.

Reply:
I'm not entirely sure the described scenario applies during "development time" use-case, but I get we want to be sure the exception is raised IFF the try clause actually ended correcly hence semantically to raise an exception on confirmation we are in the scenario of `event_handler.is_modified() is True` (that would happen when try-clause completes without other errors)

Will work into that, thanks

Reply:
implemented with c7512df